### ISSUE TITLE
multiple simultaneous chart renders causes slowdown 

### ISSUE THREAD

**Issue by davelandry** [d3plus/d3plus-plot#102]
### Expected Behavior
Re-rendering charts at the same time should not cause such slowdown.

### Current Behavior
There is exponential lag when triggering multiple chart re-renders at the same time.

### Steps to Reproduce (for bugs)
https://jsfiddle.net/ohpqz7se/23/

### Your Environment

* Browser Name: Chrome
* Operating System and Version: macOS High Sierra

**Comment by cnavarreteliz** [comment]
![image](https://user-images.githubusercontent.com/5233593/41251811-161201aa-6d89-11e8-8331-4fe2c088fe96.png)

**Comment by davelandry** [comment]
@cnavarreteliz that timer validates that d3plus is the culprit, but does it provide us any additional information?

**Comment by davelandry** [comment]
@cnavarreteliz if you clone this repository and test locally, the sourcemaps should be able to give more information

**Comment by cnavarreteliz** [comment]
Nop, only it says that `Ne()` function is the problem.

**Comment by cnavarreteliz** [comment]
@davelandry I isolated the problem commenting `tickData` in `d3plus-axis`. I'll revise more deeply that part of the code.

![verizon](https://user-images.githubusercontent.com/5233593/41255147-96b0b96e-6d93-11e8-9a25-3d25428959cf.gif)

**Comment by cnavarreteliz** [comment]
Root cause: [https://github.com/d3plus/d3plus-axis/blob/master/src/Axis.js#L669-L678](https://github.com/d3plus/d3plus-axis/blob/master/src/Axis.js#L669-L678)

**Comment by cnavarreteliz** [comment]
![verizon](https://user-images.githubusercontent.com/5233593/41258249-8d7f9cac-6d9d-11e8-9ca8-34ce7cddabf3.gif)

**Comment by davelandry** [comment]
iiiiiinteresting, great detective work here @cnavarreteliz.

maybe it's actually an inefficiency with `d3plus-shape`, and how it renders lines? each tick on an axis is just a `Line`, using the shape's built in label for the text. It's really just draw the bar shapes (~10 shapes), plus the tick shapes (probably ~20 more shapes). 30 shapes shouldn't be causing that much slowdown...

**Comment by cnavarreteliz** [comment]
@davelandry I don't have been able to solve yet the problem with slow transitions. But today I discovered that in `d3plus-text`, same label is re-render three times. Qualitatively I think that the problem is that.

I was trying to isolate this: [https://github.com/d3plus/d3plus-text/blob/master/src/TextBox.js#L251-L313](https://github.com/d3plus/d3plus-text/blob/master/src/TextBox.js#L251-L313)

**Comment by cnavarreteliz** [comment]
![image](https://user-images.githubusercontent.com/5233593/41317295-c422d100-6e62-11e8-8fb3-2c4b6a07e31a.png)

**Comment by davelandry** [comment]
@cnavarreteliz is it actually rendering 3 times? That `.each()` block draws an individual `<text>` tag for each line, all inside of a `<g>` tag. For example, a block of text that has 3 lines would look like this:

```html
<g>
  <text>This is a long</text>
  <text>sentence that is</text>
  <text>on three lines.</text>
</g>
```

**Comment by davelandry** [comment]
closed with #103
